pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Tell me if you think differently, but I'm inclined to consider the following types of conflicts a non-issue:
* The user removes an object or property that an automated process has changed.
* The user changes an object or property that an automated process has removed.
* The user changes an object or property that an automated process has changed to a different value.

While we can signal a conflict and make the user repeat the operation, forcing the change, in each of these cases, 9 times out of 10, the user will want their changes to take precedence.